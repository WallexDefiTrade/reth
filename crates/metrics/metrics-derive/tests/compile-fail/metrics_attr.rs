extern crate reth_metrics_derive;
use reth_metrics_derive::Metrics;

fn main() {}

#[derive(Metrics)]
struct CustomMetrics;

#[derive(Metrics)]
#[metrics()]
#[metrics()]
struct CustomMetrics2;
