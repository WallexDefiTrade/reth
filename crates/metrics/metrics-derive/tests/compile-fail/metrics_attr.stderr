error: `#[metrics(..)]` attribute must be provided.
 --> tests/compile-fail/metrics_attr.rs:7:1
  |
7 | struct CustomMetrics;
  | ^^^^^^^^^^^^^^^^^^^^^

error: Duplicate `#[metrics(..)]` attribute provided.
  --> tests/compile-fail/metrics_attr.rs:11:1
   |
11 | #[metrics()]
   | ^^^^^^^^^^^^
